core/flamingock-test-support/src/main/java/io/flamingock/support/validation/impl/AuditSequenceStrictValidator.java

```diff
@@ -15,33 +15,99 @@
+ * sequence exactly, both in count and in field values. Checking:</p>
+ * <ul>
+ *   <li>Exact count match between expected and actual entries</li>
+ *   <li>Strict field-by-field validation for each entry at each index</li>
+ *   <li>Order preservation (expected[0] must match actual[0], etc.)</li>
+ * </ul>
+ */
 public class AuditSequenceStrictValidator implements SimpleValidator {
 
     private static final String VALIDATOR_NAME = "Audit Sequence (Strict)";
 
-    private final AuditStore<?> auditStore;
-    private final List<AuditEntryExpectation> expectations;
-
+    private final List<AuditEntryExpectation> expectedExpectations;
+    private final List<AuditEntry> actualEntries;
 
+    /**
+     * Creates a strict sequence validator from an AuditStore and expected definitions.
+     *
+     * @param auditStore  the audit store to read actual entries from
+     * @param definitions the expected audit entry definitions
+     */
     public AuditSequenceStrictValidator(AuditStore<?> auditStore, AuditEntryDefinition... definitions) {
-        this.auditStore = auditStore;
-        this.expectations = Arrays.stream(definitions)
+        this(Arrays.asList(definitions), auditStore.getPersistence().getAuditHistory());
+    }
+
+    /**
+     * Internal constructor for direct list initialization (used by tests).
+     */
+    AuditSequenceStrictValidator(List<AuditEntryDefinition> expectedDefinitions, List<AuditEntry> actualEntries) {
+        this.expectedExpectations = expectedDefinitions.stream()
                 .map(AuditEntryExpectation::new)
                 .collect(Collectors.toList());
+        this.actualEntries = actualEntries != null ? actualEntries : new ArrayList<>();
     }
 
     @Override
     public ValidationResult validate() {
-        // TODO: Implement actual validation logic
-        return ValidationResult.success(VALIDATOR_NAME);
+        // Check count
+        if (expectedExpectations.size() != actualEntries.size()) {
+            return ValidationResult.failure(VALIDATOR_NAME,
+                    new CountMismatchError(getExpectedChangeIds(), getActualChangeIds()));
+        }
+
+        // Validate each entry in sequence
+        List<ValidationError> allErrors = getValidationErrors(expectedExpectations, actualEntries);
+
+        if (allErrors.isEmpty()) {
+            return ValidationResult.success(VALIDATOR_NAME);
+        }
+
+        return ValidationResult.failure(VALIDATOR_NAME, allErrors.toArray(
+                new io.flamingock.support.validation.error.ValidationError[0]));
+    }
+
+    private static List<ValidationError> getValidationErrors(List<AuditEntryExpectation> expectedExpectations, List<AuditEntry> actualEntries) {
+        List<ValidationError> allErrors = new ArrayList<>();
+        for (int i = 0; i < expectedExpectations.size(); i++) {
+            AuditEntryExpectation expected = expectedExpectations.get(i);
+            AuditEntry actual = actualEntries.get(i);
```

Comment:
This will raise an indexOutOfBound exception when expected when expected.size() > actualEntries.size()